src/main/java/de/unistuttgart/overworldbackend/repositories/WorldRepository.java

```diff
@@ -0,0 +1,15 @@
+package de.unistuttgart.overworldbackend.repositories;
+
+import de.unistuttgart.overworldbackend.data.World;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface WorldRepository extends JpaRepository<World, UUID> {
+  Optional<World> findByIdAndLectureId(UUID id, int lectureId);
```

Comment:
We just thought it would be strange if you request /lectures/<lectureId>/worlds/<worldId> get the same response as /lectures/<otherLectureId>/worlds/<worldId> but one of the lecture does not contain the world. So we make sure the requested world belongs to the requested lecture.